### PR TITLE
feat: Refactor button component

### PR DESCRIPTION
**Pull request by dlnr**
To slim down our bundle and allow for more of our customizations we opted to use our own button.

### CODE REVIEW

File: packages/css/src/components/button/button.scss [hunk 1]
```diff
@@ -3,47 +3,96 @@
...
  */
 
-@import "../../../node_modules/@utrecht/components/button/css";
-
-@mixin reset {
-  -webkit-text-size-adjust: 100%;
-}
-
-.amsterdam-button {
+@mixin amsterdam-button {
```
Comment (VincentSmedinga) [inline packages/css/src/components/button/button.scss:6]: What’s the added value of using mixins? A button already has both `.amsterdam-button` and `.amsterdam-button-${variant}`, so we could use classes as well. No other components use the mixins approach.

File: packages/css/src/components/button/button.scss [hunk 2]
```diff
@@ -3,47 +3,96 @@
...
     font-size: var(--amsterdam-breadcrumb-compact-font-size);
     line-height: var(--amsterdam-breadcrumb-compact-line-height);
   }
+
+  &:disabled,
+  &[aria-disabled="true"] {
+    cursor: var(--amsterdam-action-disabled-cursor);
+  }
+
+  &:hover:not(:disabled, [aria-disabled="true"]) {
```
Comment (VincentSmedinga) [inline packages/css/src/components/button/button.scss:29]: Can we lose the `:not` selector by moving this block to before the `disabled` block?

File: packages/css/src/components/button/button.scss [hunk 3]
```diff
@@ -3,47 +3,96 @@
...
+  }
+}
+
+@mixin amsterdam-button-forced-color-mode {
+  @media screen and (-ms-high-contrast: active), screen and (forced-colors: active) {
+    border: 2px solid ButtonBorder; // add border because forced colors changes box-shadow to none
+  }
+}
+
+.amsterdam-button--busy {
```
Comment (VincentSmedinga) [inline packages/css/src/components/button/button.scss:40]: I’m not seeing this in action. Does it need a `:hover` selector added? Can we add the arg to the stories?

File: packages/css/src/components/button/button.scss [hunk 4]
```diff
@@ -3,47 +3,96 @@
...
+  cursor: var(--amsterdam-action-busy-cursor);
+}
+
+.amsterdam-button--primary {
+  background-color: var(--amsterdam-button-primary-background-color);
+  color: var(--amsterdam-button-primary-color);
```
Comment (VincentSmedinga) [inline packages/css/src/components/button/button.scss:46]: Should we set `box-shadow` (with the same colour as the background) for consistency and reusability with the other variants?

File: packages/css/src/components/button/button.scss (hunk 3 above)
Comment (dlnr) [reply packages/css/src/components/button/button.scss:40]: It's usually a rendered state, for when you're waiting for other tasks to complete. I'll add some stories.

Comment (VincentSmedinga) [reply packages/css/src/components/button/button.scss:40]: Yes I mean I tested it locally, but didn’t see the cursor change.

File: packages/css/src/components/button/button.scss (hunk 4 above)
Comment (dlnr) [reply packages/css/src/components/button/button.scss:46]: Good idea

**Review by VincentSmedinga — DISMISSED** [review]

File: proprietary/tokens/src/components/amsterdam/button.tokens.json [hunk 5]
```diff
@@ -1,6 +1,13 @@
 {
   "amsterdam": {
     "button": {
+      "font-family": { "value": "{amsterdam.typography.font-family}" },
+      "gap": { "value": "{amsterdam.spacing.inset.md}" },
+      "padding-inline-start": { "value": "{amsterdam.spacing.inset.md}" },
+      "padding-inline-end": { "value": "{amsterdam.spacing.inset.md}" },
+      "padding-block-start": { "value": "{amsterdam.spacing.inset.sm}" },
+      "padding-block-end": { "value": "{amsterdam.spacing.inset.sm}" },
```
Comment (alimpens) [inline proprietary/tokens/src/components/amsterdam/button.tokens.json:9]: The spacing tokens aren't really well thought out yet, I wouldn't use them here.

File: storybook/storybook-react/src/Button/Button.docs.mdx [hunk 6]
```diff
@@ -1,15 +1,11 @@
...
 import README from "../../../../packages/css/src/components/button/README.md?raw";
 
 <Meta of={ButtonStories} />
 
 <Markdown>{README}</Markdown>
 
-<Primary />
-
-<Controls />
-
```
Comment (alimpens) [inline storybook/storybook-react/src/Button/Button.docs.mdx:12]: Why did you remove these?

File: storybook/storybook-react/src/Button/Button.stories.tsx [hunk 7]
```diff
@@ -25,16 +41,93 @@ export const Primary: Story = {
...
+export const SecondaryWithIconAndBadge: Story = {
+  args: {
+    variant: 'secondary',
+    children: [
+      'Secondary',
+      <Badge key="badge" label={99} color="dark-blue" />,
+      <Icon key="icon" svg={CloseIcon} size="level-5" />,
+    ],
+  },
+}
```
Comment (alimpens) [inline storybook/storybook-react/src/Button/Button.stories.tsx:73]: I don't think you have to show you can add other components to Button 4 times, PrimaryWithIcon is enough imo

File: packages/css/src/components/button/button.scss [hunk 8]
```diff
@@ -3,47 +3,95 @@
...
+  font-family: var(--amsterdam-button-font-family);
   font-size: var(--amsterdam-button-spacious-font-size);
+  gap: var(--amsterdam-button-gap);
   line-height: var(--amsterdam-button-spacious-line-height);
-
-  @include reset;
+  outline-offset: var(--amsterdam-button-outline-offset);
+  padding-block: var(--amsterdam-button-padding-block-start) var(--amsterdam-button-padding-block-end);
+  padding-inline: var(--amsterdam-button-padding-inline-start) var(--amsterdam-button-padding-inline-end);
+  -webkit-text-size-adjust: 100%;
```
Comment (alimpens) [inline packages/css/src/components/button/button.scss:17]: I think we want to wrap this in a reset mixin, like we did before

File: packages/css/src/components/button/button.scss [hunk 9]
```diff
@@ -3,47 +3,95 @@
...
```
Comment (alimpens) [inline packages/css/src/components/button/button.scss:18]: We're missing `touch-action: manipulation` here

File: proprietary/tokens/src/components/amsterdam/button.tokens.json (hunk 5 above)
Comment (dlnr) [reply proprietary/tokens/src/components/amsterdam/button.tokens.json:9]: They were also used in the previous button setup and they're just `1rem` and `.5rem` values.

File: storybook/storybook-react/src/Button/Button.docs.mdx (hunk 6 above)
Comment (dlnr) [reply storybook/storybook-react/src/Button/Button.docs.mdx:12]: They interupted the docs without added value.

File: storybook/storybook-react/src/Button/Button.stories.tsx (hunk 7 above)
Comment (dlnr) [reply storybook/storybook-react/src/Button/Button.stories.tsx:73]: This was left in here on purpose to test the theory that we don't need a filter button

Comment (dlnr) [reply storybook/storybook-react/src/Button/Button.stories.tsx:73]: Changed it and moved busy/disabled to controls. I kept secondary with a badge as an example for the filter button discussion. Is that ok?

Comment (alimpens) [reply storybook/storybook-react/src/Button/Button.stories.tsx]: I'm not sure we need to show a filter button with a Badge, that design doesn't seem to be very fleshed out. In most examples, a filter button is a Button with a Close Icon. That's all I would show here. Maybe we should call the story 'Button with an icon', so we don't imply that only the Primary Button can have an Icon. I'd move it to after the Tertiary Button story, so that distinction is more clear

File: packages/css/src/components/button/button.scss [hunk 10]
```diff
@@ -3,42 +3,96 @@
...
 
-@import "../../../node_modules/@utrecht/components/button/css";
-
 @mixin reset {
   -webkit-text-size-adjust: 100%;
 }
 
 .amsterdam-button {
+  border: none;
+  cursor: var(--amsterdam-action-activate-cursor);
```
Comment (alimpens) [inline packages/css/src/components/button/button.scss:12]: We shouldn't import common level tokens directly in the css. The component level tokens should import the brand/common level ones.

That way, external users only have to override the component level tokens to get their styling.

File: packages/css/src/components/button/button.scss [hunk 11]
```diff
@@ -3,42 +3,96 @@
...
+  display: inline-flex;
+  font-family: var(--amsterdam-button-font-family);
   font-size: var(--amsterdam-button-font-size);
+  gap: var(--amsterdam-button-gap);
   line-height: var(--amsterdam-button-line-height);
+  outline-offset: var(--amsterdam-button-outline-offset);
+  padding-block: var(--amsterdam-button-padding-block-start) var(--amsterdam-button-padding-block-end);
```
Comment (alimpens) [inline packages/css/src/components/button/button.scss:19]: Why do we use a shorthand here if we're going to define both values anyway? As far as I'm concerned, we only have to define 1 value here, but if we want to define both, it makes more sense to not use the shorthand here imo.

**Review by alimpens — APPROVED** [review]
